src/neighborhoods/ball.jl

```diff
@@ -37,6 +37,11 @@ metric(ball::BallNeighborhood) = ball.metric
 isneighbor(ball::BallNeighborhood, xₒ::AbstractVector, x::AbstractVector) =
   evaluate(ball.metric, xₒ, x) ≤ ball.radius
 
+function EllipsoidNeighborhood(semiaxes, angles; convention=:TaitBryanExtr)
+  metric = aniso2distance(semiaxes, angles, convention=convention)
+  BallNeighborhood(1.0, metric)
+end
+
```

Comment:
What do you think of moving this to a separate source file `ellipsoid.jl` in the same folder? I know it is just a helper function, but maybe we could place it in a separate file and write a docstring for it pointing to the `aniso2distance` docstring.

Reply:
So the question of moving to a separate file can only be addressed after we brainstorm possible use cases where it is important to differentiate between `EllipsoidNeighborhood` and `BallNeighborhood`. If it is important to different and dispatch different methods, we need a separate type `EllipsoidNeighborhood` in a separate source file. It if it is not important, I can leave the definition here as a helper function that returns a `BallNeighborhood`.

Reply:
I was initially inclined to think that it needed to be a separate structure. But now I think it's really just a `BallNeighborhood` with different metric, where the helper function is useful to simplify the call. The restrictions can also be applied to both, including octants. I'm just not totally sure if I can make the octants restriction using the `Mahalanobis.qmat` as it is (I need to transform the coordinates in order that the semiaxes match xyz axis).

Reply:
Yes, I also cannot see a case where we would need to differentiate the behavior of `EllipsoidNeighborhood` from the behavior of `BallNeighborhood`, so I think we are fine with the helper function like this. We just need a docstring and tests 👍🏽 

If you can describe the octant restriction that would help understand your concerns with coordinates transformations.